MyApplication/app/src/main/java/com/jonathanlieblich/myapplication/MainActivity.java

```diff
@@ -0,0 +1,108 @@
+package com.jonathanlieblich.myapplication;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.Button;
+import android.widget.Toast;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class MainActivity extends AppCompatActivity {
+    public static final String URL_START =
+            "http://api.walmartlabs.com/v1/search?apiKey=s4x4jthz6j87wwmd5s3vuafd&query=";
+    public static final String CEREAL = "cereal";
+    public static final String CHOCOLATE = "chocolate";
+    public static final String TEA = "tea";
+    private Button mCereal, mChocolate, mTea;
+    private RecyclerView mRecycler;
+    private ProductViewAdapter mAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mCereal = (Button)findViewById(R.id.cereal);
+        mChocolate= (Button)findViewById(R.id.chocolate);
+        mTea = (Button)findViewById(R.id.tea);
+
+        View.OnClickListener listener = new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                switch(view.getId()) {
```

Comment:
Here at the beginning of `onClick()` would be the place to check for internet connectivity